### PR TITLE
Streamline option naming for several processors

### PR DESCRIPTION
**Pull request by martijnvg**
- For `rename` processor, renamed `to` to `target_field`
- For `date` processor, renamed `match_field` to `field` and `match_formats` to `formats`.
- For `geoip` processor, renamed `source_field` to `field` and renamed `fields` to `properties`
- For `attachment` processor, renamed `source_field` to `field` and renamed `fields` to `properties`

PR for #17835

### CODE REVIEW

File: core/src/main/java/org/elasticsearch/ingest/processor/DateProcessor.java [hunk 1]
```diff
@@ -112,7 +112,7 @@ String getTargetField() {
 
         @SuppressWarnings("unchecked")
         public DateProcessor doCreate(String processorTag, Map<String, Object> config) throws Exception {
-            String matchField = ConfigurationUtils.readStringProperty(TYPE, processorTag, config, "match_field");
+            String field = ConfigurationUtils.readStringProperty(TYPE, processorTag, config, "field");
             String targetField = ConfigurationUtils.readStringProperty(TYPE, processorTag, config, "target_field", DEFAULT_TARGET_FIELD);
             String timezoneString = ConfigurationUtils.readOptionalStringProperty(TYPE, processorTag, config, "timezone");
             DateTimeZone timezone = timezoneString == null ? DateTimeZone.UTC : DateTimeZone.forID(timezoneString);
```
Comment (javanna) [inline core/src/main/java/org/elasticsearch/ingest/processor/DateProcessor.java]: we may want to rename match_formats as well here, can do in another PR though.

**Comment by javanna** [comment]
left one tiny comment. I am not too sure about `properties` but I can't come up with anything better, maybe @clintongormley wants to comment on that. LGTM otherwise

File: core/src/main/java/org/elasticsearch/ingest/processor/DateProcessor.java (hunk 1 above)
Comment (martijnvg) [inline core/src/main/java/org/elasticsearch/ingest/processor/DateProcessor.java]: I thought about when changing this file, so just rename it too `formats`?

Comment (javanna) [inline core/src/main/java/org/elasticsearch/ingest/processor/DateProcessor.java]: ++

**Comment by martijnvg** [comment]
@javanna I've updated the pr and renamed `match_formats` to `formats`.

**Comment by javanna** [comment]
LGTM thanks @martijnvg let's see what Clint thinks about the `properties` naming?

**Comment by clintongormley** [comment]
`properties`++
